app/grandchallenge/algorithms/models.py

```diff
@@ -241,7 +241,7 @@ class Meta(UUIDModel.Meta, TitleSlugDescriptionModel.Meta):
         ]
 
     def __str__(self):
-        return f"{self.title}"
+        return f"{self.title} (Active version: {self.active_image})"
```

Comment:
Did you check for N+1 problems on all the dropdowns where this is used?

Reply:
Ah good point. There is an N+1 problem with this indeed, but somehow prefetching `algorithm_container_images` in the respective view(s) doesn't help. Have to look into it more.

Reply:
This is strange. Maybe it has to do with the fact that we filter the `algorithm_container_images` to get the active image?

Reply:
It is because `active_image` does a `.get()` for each row. This is fine as we're usually only doing this for one algorithm. I think it would be best if this was removed from the `__str__` method and we tried another approach as I think that is really difficult to solve with all our rules.

We do this weird thing where we get the algorithm in the form and then set the algorithm image from that (https://github.com/comic/grand-challenge.org/blob/255996b8f373d391cac93529e40bbbe89901b6ab/app/grandchallenge/evaluation/views/__init__.py#L296-L300), I wonder if we should change things around so that we get the algorithm image directly instead?

Reply:
Ah that's a good idea. We don't have a manager method for that yet, but that can be easily added.